### ISSUE TITLE
Varia crashes on startup with reason "no attribute 'USER_DIRECTORY_HOME'"

### ISSUE THREAD

**Issue by Nydragon** [giantpinkrobots/varia#107]
Varia crashes on startup with the following error: `AttributeError: 'gi.repository.GLib' object has no attribute 'USER_DIRECTORY_HOME'`.

<details><summary>Full Error</summary>
<p>

```
Traceback (most recent call last):
  File "/app/share/varia/varia/variamain.py", line 379, in main
    if (os.path.exists(GLib.get_user_special_dir(GLib.USER_DIRECTORY_DOWNLOAD))):
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "<frozen genericpath>", line 19, in exists
TypeError: stat: path should be string, bytes, os.PathLike or integer, not NoneType

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/app/share/varia/../../bin/varia-py.py", line 36, in <module>
    sys.exit(main(VERSION, aria2cexec))
             ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/app/share/varia/varia/variamain.py", line 384, in main
    download_directory = GLib.get_user_special_dir(GLib.USER_DIRECTORY_HOME)
                                                   ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/site-packages/gi/overrides/__init__.py", line 31, in __getattr__
    return getattr(self._introspection_module, name)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/site-packages/gi/module.py", line 130, in __getattr__
    raise AttributeError("%r object has no attribute %r" % (
AttributeError: 'gi.repository.GLib' object has no attribute 'USER_DIRECTORY_HOME'
```

</p>
</details> 

I am on NixOS with Sway. I tried the NixOS package (2024.3.20) and flatpak (2024.5.7) one and they both had the exact same error.

**Comment by Nydragon** [comment]
I have the following XDG directory env variables set
```
XDG_SEAT=
XDG_DESKTOP_PORTAL_DIR=
XDG_CURRENT_DESKTOP=
XDG_ACTIVATION_TOKEN=
XDG_DATA_DIRS=
XDG_CACHE_HOME=
XDG_RUNTIME_DIR=
XDG_SESSION_TYPE=
XDG_CONFIG_DIRS=
XDG_SESSION_ID=
XDG_CONFIG_HOME=
XDG_VTNR=
XDG_STATE_HOME=
XDG_DATA_HOME=
XDG_SESSION_CLASS=
```

**Comment by N0tACyb0rg** [comment]
Could you try the fix suggested from #94?

**Comment by Nydragon** [comment]
It worked, thanks a lot!
